### ISSUE TITLE
/api/blocks not working correctly >= 3.2.1

### ISSUE THREAD

**Issue by alessiodf** [ArkEcosystem/core#4640]
Since Core 3.2.1, the `/api/blocks` endpoint is not working correctly. I initially reported to @dated that the default sorting method is no longer descending by height (which is still the case in 3.2.2 too), but I see there are more problems so I am posting it as an issue now.

But more importantly, some blocks are simply missing from the results. Take, for example, the current output of https://api.ark.io/api/blocks?limit=5 (prettified and limited to 5 for ease of reading below). It skips from 4115424 to 4115426 (missing 4115425), 29, 31, 33 and so on. It occurs regardless of `limit` value and even if no `limit` is specified.

The issue is not present in 3.2.0. 

```
{
  "meta": {
    "totalCountIsEstimate": false,
    "count": 5,
    "pageCount": 3958220,
    "totalCount": 19791100,
    "next": "/blocks?limit=5&transform=true&page=2",
    "previous": null,
    "self": "/blocks?limit=5&transform=true&page=1",
    "first": "/blocks?limit=5&transform=true&page=1",
    "last": "/blocks?limit=5&transform=true&page=3958220"
  },
  "data": [
    {
      "id": "7753667336870923744",
      "version": 0,
      "height": 4115424,
      "previous": "17081634046000381964",
      "forged": {
        "reward": "200000000",
        "fee": "0",
        "amount": "0",
        "total": "200000000"
      },
      "payload": {
        "hash": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
        "length": 0
      },
      "generator": {
        "username": "jamiec79",
        "address": "ARKfan7iMEK3cn8QDWVzcKdCCpL4MkASHb",
        "publicKey": "025631d7f863d2688443a14654e1b308c7585cb5f78563b3e785a978fdefff2c3e"
      },
      "signature": "304402202e85fba6c8b27f89e9e61a0700b259003c5f937e771cb8d286321a6296948b0f0220068d73d8fc327acb21e05f8820c6b2c0b32f329ecfa28a3720a6e8816582ea5d",
      "confirmations": 15675676,
      "transactions": 0,
      "timestamp": {
        "epoch": 33356136,
        "unix": 1523457336,
        "human": "2018-04-11T14:35:36.000Z"
      }
    },
    {
      "id": "8737217756359020838",
      "version": 0,
      "height": 4115426,
      "previous": "3255648078916117029",
      "forged": {
        "reward": "200000000",
        "fee": "0",
        "amount": "0",
        "total": "200000000"
      },
      "payload": {
        "hash": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
        "length": 0
      },
      "generator": {
        "username": "superstar",
        "address": "AUrJhHwS6KLj9Q9dQdamALdjpco5v8jw2T",
        "publicKey": "02a6b88567f86fa22d3c958f470c1645b3f65aa555e00c5bb0eecd2886eab80d34"
      },
      "signature": "3045022100f80b203e3c166d58beea53c3a91ec85464f7f43a45a6762b3bb361497cdac9be02203ec4765f97fc85953219d6b528d7303e0c47475687921b0c08a95f489bb0af06",
      "confirmations": 15675674,
      "transactions": 0,
      "timestamp": {
        "epoch": 33356152,
        "unix": 1523457352,
        "human": "2018-04-11T14:35:52.000Z"
      }
    },
    {
      "id": "9909239301628456784",
      "version": 0,
      "height": 4115429,
      "previous": "11731906316427906939",
      "forged": {
        "reward": "200000000",
        "fee": "0",
        "amount": "0",
        "total": "200000000"
      },
      "payload": {
        "hash": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
        "length": 0
      },
      "generator": {
        "username": "tibonos",
        "address": "AMdPL5Br3nsWyJB4LCB5qZ9cYnKBdVXKJi",
        "publicKey": "02bd74ba354e4533cbbb0d03076622c15917e2e1d65fe4c4c8294d6b4c57d667b4"
      },
      "signature": "30440220255da2dee083c35f9739b33ae98b3eaa7e4cd76578ed3097185aeca8dfc1d293022056d279ece981ccd8cef966e93b144384ecec2f6385e14baa46953cef3fa071ba",
      "confirmations": 15675671,
      "transactions": 0,
      "timestamp": {
        "epoch": 33356176,
        "unix": 1523457376,
        "human": "2018-04-11T14:36:16.000Z"
      }
    },
    {
      "id": "7840555788618904812",
      "version": 0,
      "height": 4115431,
      "previous": "7076165191716698861",
      "forged": {
        "reward": "200000000",
        "fee": "0",
        "amount": "0",
        "total": "200000000"
      },
      "payload": {
        "hash": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
        "length": 0
      },
      "generator": {
        "username": "doc",
        "address": "AR2iXnLRp9HuoT584m6tjNWLHGWdX8uuGK",
        "publicKey": "03cefbfa0c1c853084591b62a9aad0116029eaebdc27c2e3b811b1b0aebb928fc6"
      },
      "signature": "3045022100bb81933c4ba280af5c984205a463126f14fe7e5beebf39b1f55e9d55d37343420220089ba438ccd2a40f74c380551a51c41ee168dc60a59e98b377d0993fd3af4468",
      "confirmations": 15675669,
      "transactions": 0,
      "timestamp": {
        "epoch": 33356192,
        "unix": 1523457392,
        "human": "2018-04-11T14:36:32.000Z"
      }
    },
    {
      "id": "16671560697211721601",
      "version": 0,
      "height": 4115433,
      "previous": "18210158330549674154",
      "forged": {
        "reward": "200000000",
        "fee": "0",
        "amount": "0",
        "total": "200000000"
      },
      "payload": {
        "hash": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
        "length": 0
      },
      "generator": {
        "username": "dafty",
        "address": "AdAftY6ZZQ4h6CoKgVDyY4ms2rGFzpHDPT",
        "publicKey": "035845a4f1125ce10f4cbe8a271378b0004c5722874a72dc214044107e4b21b324"
      },
      "signature": "3045022100f69e2200d2ba0bb44a05373add66039a3f2d122f866c5b6c84ca3116da0f98db022058597513ce3645deba086e55c5de92558b7bcb3de7e35e614c91b5034bf64ecd",
      "confirmations": 15675667,
      "transactions": 0,
      "timestamp": {
        "epoch": 33356208,
        "unix": 1523457408,
        "human": "2018-04-11T14:36:48.000Z"
      }
    }
  ]
}
```

**Comment by crnazmija** [comment]
Can confirm there's something weird with default block sorting.
First time I got heights: 4817972, 4817974, 4817976, 4817838, 4817840. Ascending with missing blocks.
All other tries I got: 9252193, 9252194, 9252195, 9252196, 9252197. Ascending with no missing blocks.
Weird height to start from in both cases.

Note, it works as expected with an explicit sort parameter: https://api.ark.io/api/blocks?limit=5&orderBy=height:desc
